### DISCUSSION TITLE
Running gpth dircectly on Synology

### DISCUSSION THREAD

**Discussion by Pairis** [TheLastGimbus/GooglePhotosTakeoutHelper#233] [I NEED HELP]
Hi @TheLastGimbus,

I have little to no experience when it comes to Terminal, but I keep encountering some issues and hope you can help me out!

The Takeout folder lives on my Synology NAS. It all works as expected when I execute the gpth from my Mac. But as it's such a large database, the Mac does not keep the connection stable over time resulting in an error and termination of the program.

Can I run it locally on the NAS? Or do you have an alternative suggestion to make sure the gpth finishes without me having to be active all the time?

Big thanks!

**Comment by TheLastGimbus** [comment]
Yess, look up to #157, those guys figured it all out ^_^

But tl;dr you will just need to specify everything with cli arguments instead of interactive, but you can do it! 

type in `gpth --help` in the terminal and you will see all of them:

```
gpth --help

[...]

Then, run: gpth --input "folder/with/all/takeouts" --output "your/output/folder"
...and gpth will parse and organize all photos into one big chronological folder

-h, --help                        
    --fix                         Folder with any photos to fix dates. This skips whole "GoogleTakeout" procedure.It is here because gpth has some cool heuristics to determine date of a photo, and this can be handy in many situations :)
    --[no-]interactive            Use interactive mode. Type this in case auto-detection fails, or you *really* want to combine advanced options with prompts
-i, --input                       Input folder with *all* takeouts *extracted*. 
-o, --output                      Output folder where all photos will land
    --albums                      What to do about albums?

          [duplicate-copy]        Album folders with photos copied into them. This will work across all systems, but may take wayyy more space!!
          [json]                  Put ALL photos (including Archive and Trash) in one folder and make a .json file with info about albums. Use if you're a programmer, or just want to get everything, ignoring lack of year-folders etc.
          [nothing]               Just ignore them and put year-photos into one folder. WARNING: This ignores Archive/Trash !!!
          [shortcut] (default)    [Recommended] Album folders with shortcuts/symlinks to original photos. Recommended as it will take the least space, but may not be portable when moving across systems/computes/phones etc

    --[no-]divide-to-dates        Divide output to folders by year/month
    --[no-]skip-extras            Skip extra images (like -edited etc)
    --[no-]guess-from-name        Try to guess file dates from their names
                                  (defaults to on)
    --[no-]copy                   Copy files instead of moving them.
                                  This is usually slower, and uses extra space, but doesn't break your input folder
```

what *really* interests you is the `--input` and `--output` ;)

**Reply by Pairis** [reply]
Thanks for your reply. 

I just figured out how to 'SSH' my NAS with[ this guide](https://www.synology-forum.nl/algemeen/nas-benaderen-met-ssh-winscp-putty/). Next pice of the puzzle is how to run gpth on the NAS as I get: 
`chmod: cannot access 'gpth': No such file or directory`

I'll scroll through the topic you've suggested.

**Reply by TheLastGimbus** [reply]
Well... do you have a binary named `gpth` there... ? Or it's gpth-linux, or nothing yet :see_no_evil:

**Comment by Pairis** [comment]
I managed to run gpth from my mac during a work-from-home-day. That way I did not lose connection. 

Really happy with the results! Thanks for developing! 

I'll need to check your instructions regarding dates. I've got about 10K/70K images where it could not find the date.

**Reply by TheLastGimbus** [reply]
If you show me screenoshot of those images we may find a way to guess them
